### PR TITLE
112 last contacted from dynamodb

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

File: Speaker-comms/LastContactedService.fs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿module Speakercomms.LastContactedService
```
Comment (csscottlogic) [inline Speaker-comms/LastContactedService.fs]: Should be ```SpeakerComms``` with an upper case C?

File: Speaker-comms/LastContactedService.fs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿module Speakercomms.LastContactedService
+
+open Speakercomms.CorrespondenceService
+open System
+open Speakercomms.Models
+
+let updateMap key value (map : Map<String, String>) = 
+    if not <| map.ContainsKey(key) || ((Map.find key map) < value)then
```
Comment (csscottlogic) [inline Speaker-comms/LastContactedService.fs]: space before the ```then```

File: Speaker-comms/LastContactedService.fs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+let updateMap key value (map : Map<String, String>) = 
+    if not <| map.ContainsKey(key) || ((Map.find key map) < value)then
+        map.Add (key, value)
+    else 
+        map
+
+let getLastContacted () =                     
+    let items = getCorrespondence()
+    let mutable lastContactedMap : Map<String, String> = Map.empty
```
Comment (csscottlogic) [inline Speaker-comms/LastContactedService.fs]: This is an example of where local mutability would make things easier. You're effectively shifting the mutability to the reference to the map and replacing the entire map every time. A Dictionary<string, string> would be fine.
Also, the mutable dictionary is being returned from this and not returned at a module level, so there are no shared references. If the caller then decides to mutate what it's been given, there won't be any problems.

File: Speaker-comms/Models.fs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿module Speakercomms.Models
+
+open System
+open Amazon.DynamoDBv2.DataModel
+
+[<DynamoDBTable("Correspondence")>]
+type CorrespondenceItem () =
```
Comment (csscottlogic) [inline Speaker-comms/Models.fs]: That's nice and succinct!

File: Speaker-comms/LastContactedService.fs (hunk 3 above)
Comment (csscottlogic) [inline Speaker-comms/LastContactedService.fs]: Or turn it into an immutable map just before returning :-)

**Comment by csscottlogic** [comment]
:shipit:

File: Speaker-comms/LastContactedService.fs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+﻿module SpeakerComms.LastContactedService
```
Comment (jebbinSCL) [inline Speaker-comms/LastContactedService.fs]: I think we might decide to remove the assembly name from the module and namespace for speakercomms and speakers in the future. Good for now though

**Comment by jebbinSCL** [comment]
:shipit:

**Pull request merged**